lxd/instance.go

```diff
@@ -416,7 +416,68 @@ func instanceLoadNodeProjectAll(ctx context.Context, s *state.State, project str
 	return instances, nil
 }
 
-func autoCreateInstanceSnapshotsTask(d *Daemon) (task.Func, task.Schedule) {
+func autoCreateInstanceSnapshots(ctx context.Context, d *Daemon, instances []instance.Instance) error {
+	// Make the snapshots.
+	for _, inst := range instances {
+		ch := make(chan error)
```

Comment:
I reaslise this is just a copy/paste and not your code, but, as a separate commit, I think we have an opportunity to improve this code.

1. Although the `ch` is a channel of `error`s, it always just has nil posted to it, and is infact just used as an indicator that the go routine has ended. So instead I suggest we change this to a `struct{}` type channel and just `defer close(ch)` at the top of the go routine.
2. I'm not sure we should return from this function if part-way through a snapshot when the context is cancelled, instead we should finish the ongoing snapshot and then return. So rather than including the context's Go routine in the select, instead add a check for `ctx.Error() != nil` at the start of each for loop iteration and return if the context has been cancelled.